src/briefcase/commands/base.py

```diff
@@ -967,17 +1001,16 @@ def update_cookiecutter_cache(self, template: str, branch="master"):
                 except IndexError as e:
                     # No branch exists for the requested version.
                     raise TemplateUnsupportedVersion(branch) from e
-            except self.tools.git.exc.NoSuchPathError:
-                # Template cache path doesn't exist.
-                # Just use the template directly, rather than attempting an update.
-                cached_template = template
-            except self.tools.git.exc.InvalidGitRepositoryError:
-                # Template cache path exists, but isn't a git repository
-                # Just use the template directly, rather than attempting an update.
-                cached_template = template
-            except ValueError as e:
+            except (ValueError, self.tools.git.exc.GitError) as e:
                 raise BriefcaseCommandError(
-                    f"Git repository in a weird state, delete {cached_template} and try briefcase create again"
+                    "Unable to check out template branch.\n"
+                    "\n"
+                    "This may be because your computer is offline, or because the template repository\n"
+                    "is in a weird state. If you have a stable network connection, try deleting:\n"
+                    "\n"
+                    f"    {cached_template}\n"
+                    "\n"
+                    "retrying your command."
```

Comment:
```suggestion
                    "Unable to check out template branch.\n"
                    "\n"
                    "This may be because your computer is offline, or because the template repository\n"
                    "is in a weird state. If you have a stable network connection, try deleting:\n"
                    "\n"
                    f"    {cached_template}\n"
                    "\n"
                    "and retrying your command."
```